### PR TITLE
ci: upload the app to Play Store (WPB-8729)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8729" title="WPB-8729" target="_blank">TaskWPB-8729</a>  [Android] Setup pipeline step to upload app to beta and prod tracks
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

A Github action used here to upload the apk to Play Store based on build type.

Beta build: we upload the aab to internal
Prod build: we upload the aap to prod track with visibility of 10%
RC build: we upload it to alpha track

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/build-rc-app.yml [hunk 1]
```diff
@@ -71,6 +71,16 @@ jobs:
           KEYPWD_COMPAT: ${{ secrets.SIGNING_KEY_PASSWORD_INTERNAL_RELEASE }}
           KEYSTOREPWD_COMPAT: ${{ secrets.SIGNING_STORE_PASSWORD_INTERNAL_RELEASE }}
           ENABLE_SIGNING: ${{ secrets.ENABLE_SIGNING }}
+      - name: Create service_account.json
+        run: echo '${{ secrets.SERVICE_ACCOUNT_JSON }}' > service_account.json
+      - name: Deploy to alpha track
+        uses: r0adkll/upload-google-play@v1
+        with:
+          serviceAccountJson: service_account.json
+          packageName: com.wire.internal
```
Comment (vitorhugods) [inline .github/workflows/build-rc-app.yml]: Can't we just throw away the green app?
It's useless, isn't it?

I think absolutely none is using it and it's just one extra bloat in our configs.

**Review by vitorhugods — APPROVED** [review]
LGTM. Just thinking out loud if we really need the green app.

**Comment by AndroidBob** [comment]
Build [5094](https://10.10.124.134/job/AR-build-pipeline/5094/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9817670)(9817670)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9817670)(9817670)-staging-compat.apk)

File: .github/workflows/build-rc-app.yml (hunk 1 above)
Comment (ohassine) [reply .github/workflows/build-rc-app.yml]: Yes not used, but my idea is to use it to get playtest session build instead of using Google Drive

**Review by borichellow — APPROVED** [review]

File: .github/workflows/build-rc-app.yml (hunk 1 above)
Comment (vitorhugods) [reply .github/workflows/build-rc-app.yml]: The problem with playtest, I guess, is that some times we do it from one branch, some times from another...

Not sure how to configure that right now.

File: .github/workflows/build-prod-app.yml [hunk 2]
```diff
@@ -70,3 +70,15 @@ jobs:
...
           path: app/build/outputs/
+      - name: Create service_account.json
+        run: echo '${{ secrets.SERVICE_ACCOUNT_JSON }}' > service_account.json
+      - name: Deploy to production track
+        uses: r0adkll/upload-google-play@v1
+        with:
+          serviceAccountJson: service_account.json
+          packageName: com.wire
+          releaseFiles: app/build/outputs/bundle/prodCompatRelease/*.aab
+          track: prod
```
Comment (MohamadJaara) [inline .github/workflows/build-prod-app.yml]: if we are trying to keep the same behavior as Jenkins then please do not deploy directly onto prod track, afaik Jenkins uses Alpha track

**Review by MohamadJaara — CHANGES_REQUESTED** [review]

File: .github/workflows/build-beta-app.yml [hunk 3]
```diff
@@ -76,3 +76,14 @@ jobs:
...
+      - name: Create service_account.json
+        run: echo '${{ secrets.SERVICE_ACCOUNT_JSON }}' > service_account.json
+      - name: Deploy to internal track
+        uses: r0adkll/upload-google-play@v1
+        with:
+          serviceAccountJson: service_account.json
+          packageName: com.wire.android.internal
+          releaseFiles: app/build/outputs/bundle/betaRelease/*.aab
+          track: internal
+          mappingFile: app/build/outputs/mapping/betaRelease/mapping.txt
```
Comment (MohamadJaara) [inline .github/workflows/build-beta-app.yml]: the mapping files are not needed when uploading an aab

File: .github/workflows/build-rc-app.yml (hunk 1 above)
Comment (ohassine) [reply .github/workflows/build-rc-app.yml]: Removing it then

**Comment by AndroidBob** [comment]
Build [5122](https://10.10.124.134/job/AR-build-pipeline/5122/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9845200)(9845200)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9845200)(9845200)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]
❤️  good to have this one step closer to finish the migration

**Pull request merged**